couchdb/client.py

```diff
@@ -775,10 +775,161 @@ def put_attachment(self, doc, content, filename=None, content_type=None):
         }, rev=doc['_rev'])
         doc['_rev'] = data['rev']
 
+    def find(self, mango_query, wrapper=None):
+        """Execute a mango find-query against the database.
+
+        Note: Only available for CouchDB version >= 2.0.0 .
+
+        More information on the `mango_query` structure can be found here:
+          http://docs.couchdb.org/en/master/api/database/find.html#find-selectors
+
+        >>> server = Server()
+        >>> db = server.create('python-tests')
+        >>> db['johndoe'] = dict(type='Person', name='John Doe')
+        >>> db['maryjane'] = dict(type='Person', name='Mary Jane')
+        >>> db['gotham'] = dict(type='City', name='Gotham City')
+        >>> mango = {'selector': {'type': 'Person'},
+        ...          'fields': ['name'],
+        ...          'sort':[{'name': 'asc'}]}
+        >>> for row in db.find(mango):                          # doctest: +SKIP
+        ...    print(row['name'])                               # doctest: +SKIP
+        John Doe
+        Mary Jane
+        >>> del server['python-tests']
+
+        :param mango_query: a dictionary describing criteria used to select
+                            documents
+        :param wrapper: an optional callable that should be used to wrap the
+                        resulting documents
+        :return: the query results as a list of `Document` (or whatever `wrapper` returns)
+        """
+        assert isinstance(mango_query, dict)
```

Comment:
Is there any specific reason we need the asserts? Having these type asserts in there feels rather unidiomatic to me, what if the caller is passing some `dict`-like type that is supported in by the `Resource` class?